### DISCUSSION TITLE
Translation tools

### DISCUSSION THREAD

**Discussion by lachudra87** [MobileNativeFoundation/discussions#29] [General]
Question is simple. What tool do you use for providing texts from application to translators? At Circle K we use LingoHub and neither developers nor translators are happy about it.

Translations are very complex topic and finding a tool that will be easy to integrate with application code and in the same time have possibility to provide context of usage of each text so translators can properly translate seems to be almost impossible. 

In my previous company (big mobile devices manufacturer), special platform was created for it, but not all company can afford to write custom tool.

**Comment by Huralnyk** [comment]
Currently, we are using [Localize.io](https://localize.io) service. Previously used [POEditor](https://poeditor.com). Both platforms have similar functionality, just a matter of price and taste.

**Comment by dbarden** [comment]
I used [Phrase](https://phrase.com) in a project and was quite happy with it. We had some workflow such as who should create new keys (as in were devs allowed to do it or or should the design team handle it) and how to maintain consistency between the keys among iOS and Android, etc.

But the tool itself was quite ok. We also used screenshots heavily to show where the string would be used as well.

**Comment by BalestraPatrick** [comment]
We currently use an external vendor for all our translation tooling and for all apps (mobile and non). We have various CI configurations to push our English strings (which are the default strings) to our vendor through their API. We then have recurring jobs to sync the strings back to the repo as they are translated. All the strings are checked-in for ease of testing and completeness, so I'm curious to hear the experience of companies that are downloading them on the fly for builds and not checking them in into version control.

We have run into some problems where strings took way too long to be translated in all languages, and some strings would end up be missing in releases. Depending on how important those strings are, we had to cancel some releases to due this problem. Currently, we don't have a clear tool for teams to verify that all strings are bundled in the app for all languages before release, so this is an ad-hoc task that teams have to keep an eye on, especially when big changes are merged right before releases are cut.

**Reply by lachudra87** [reply]
on Android you can always use lint for that, it finds all not translated strings

**Reply by kgellci** [reply]
@BalestraPatrick What does your release process look like? At Lyft, we have a guarantee that if you merge your feature in before the release branch is cut, we will have translations checked into the release branch within the next 2 days, which is enough time to do validation / testing before release. 

We additionally have tooling that checks every English string has a translated counterpart for all languages that we support. The tooling runs automatically after translations are distributed and fires an alarm if any translations are missing.

Some more info here: https://github.com/MobileNativeFoundation/discussions/discussions/10

**Comment by kgellci** [comment]
We use Smartling at Lyft, although we have our own tooling for syncing the strings between the different platforms and Smartling. Posted some more info about it here: https://github.com/MobileNativeFoundation/discussions/discussions/10
